packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -660,13 +630,33 @@ export class AngularCompilerPlugin {
         }
       }
 
+      let ngccProcessor: NgccProcessor | undefined;
+      if (this._compilerOptions.enableIvy) {
+        let ngcc: typeof import('@angular/compiler-cli/ngcc') | undefined;
+        try {
+          // this is done for the sole reason that @ngtools/webpack
+          // support versions of Angular that don't have NGCC API
+          ngcc = require('@angular/compiler-cli/ngcc');
+        } catch {
+        }
+
+        if (ngcc) {
+          ngccProcessor = new NgccProcessor(
+            ngcc,
+            this._mainFields,
+            compilerWithFileSystems.inputFileSystem,
+          );
+        }
```

Comment:
Why have this block outside the try-catch? It could be much simpler:

```ts
let ngccProcessor: NgccProcessor | undefined;
if (this._compilerOptions.enableIvy) {
  try {
    // this is done for the sole reason that @ngtools/webpack
    // support versions of Angular that don't have NGCC API
    const ngcc: typeof import('@angular/compiler-cli/ngcc') | undefined = require('@angular/compiler-cli/ngcc');
    ngccProcessor = new NgccProcessor(
      ngcc,
      this._mainFields,
      compilerWithFileSystems.inputFileSystem,
    );
  }
} catch {
}
```

Reply:
As like this if NgccProcessor throws an error during it's initialisation it will be swallowed.

Reply:
OK OK